src/DocumentEditor.ts

```diff
@@ -0,0 +1,160 @@
+        if (this.recoveredFileName) {
+            await fse.ensureDir(this.recoveredDocsFolder);
+            await fse.move(this.localDocPath, path.join(this.recoveredDocsFolder, this.recoveredFileName));
+        } else {
+            await fse.unlink(this.localDocPath);
+        }
+    }
+
+    private async promptForRecoveredDocs(): Promise<void> {
+        const docs: string[] = await fse.readdir(this.recoveredDocsFolder);
+        for (const fileName of docs) {
+            const recoveredFilePath: string = path.join(this.recoveredDocsFolder, fileName);
+            const textDocument: vscode.TextDocument = await vscode.workspace.openTextDocument(recoveredFilePath);
+            await vscode.window.showTextDocument(textDocument);
+
+            const message: string = `The data in "${fileName}" may not have been updated to your CosmosDB account.`;
+            const saveFile: string = 'Save File';
+            const deleteFile: string = 'Delete File';
+            const result: string | undefined = await vscode.window.showWarningMessage(message, saveFile, deleteFile);
+            if (result === saveFile) {
+                const savedPath: vscode.Uri | undefined = await vscode.window.showSaveDialog({ filters: { JSON: ['json'] } });
+                if (savedPath) {
+                    await fse.move(recoveredFilePath, savedPath.fsPath);
+                }
+            } else if (result === deleteFile) {
+                await fse.unlink(recoveredFilePath);
+            }
+        }
+    }
+
+    private async udpateDocumentToNode(node: IDocumentNode, doc: vscode.TextDocument): Promise<void> {
+        const updatedDoc: {} = await node.update(JSON.parse(doc.getText()));
+        await this.updateEditor(updatedDoc);
+    }
+
+    private async updateEditor(data: {}): Promise<void> {
+        if (this.isLocalDocOpen && this.localDocEditor) {
+            await util.writeToEditor(this.localDocEditor, JSON.stringify(data, null, 2));
+
+            this.ignoreSave = true;
+            try {
+                await this.localDocEditor.document.save();
+            } finally {
+                this.ignoreSave = false;
+            }
+        }
+    }
+
+    private async onDidSaveTextDocument(globalState: vscode.Memento, doc: vscode.TextDocument): Promise<void> {
+        if (!this.ignoreSave && this.isLocalDocPath(doc) && this.lastOpenedDocNode) {
+            // soft-copy the node to avoid race conditions
+            const node: IDocumentNode = this.lastOpenedDocNode;
+
+            // If the user saved the file as a part of closing VS Code, we might not be able to 'Update to Azure' in time
+            // However, we can copy the file to this path and prompt user about it on next activation
+            this.recoveredFileName = `${this.lastOpenedDocNode.label}${Date.now().toString()}.json`;
+            try {
+                const dontShow: boolean | undefined = globalState.get(this.dontShowKey);
+                if (dontShow !== true) {
```

Comment:
Is this different from if(!dontShow) ? I'm assuming you're checking for dontShow being false or undefined. 
Also, how is having dontShow be false different from undefined?

Reply:
There's not a big difference. I could do it either way, but I wanted to be explicit since we're getting the setting from globalState.

For example if the globalState somehow outputs a string "false", then `if(!dontShow)` would evaluate to true